105 theme api

shchegol opened the pull request:
Не бейте сильно, в БД я не силён

+ добавлена модель siteTheme
+ добавлена модель userTheme
+ добавлены ручки для themes и userThemes
* реорганизованы routes
* bugfix userFetch при смене темы
* bugfix leaderboardAPI axios

close #105

shchegol committed:
105 theme api
+ добавлена модель siteTheme
+ добавлена модель userTheme
+ добавлены ручки для themes и userThemes
* реорганизованы routes
* bugfix userFetch при смене темы
* bugfix leaderboardAPI axios

zemer commented on src/server/models/userTheme.ts:
```diff
@@ -0,0 +1,34 @@
+
+  @Column({
+    type: DataTypes.STRING,
+    primaryKey: true,
```
primaryKey получается второй в таблице. У тебя составной ключ?

shchegol replied:
Это ошибка

shchegol committed:
#105 theme API
* bugfix повторяющийся primaryKey в UserTheme модели

alexch365 commented on src/server/models/userTheme.ts:
```diff
@@ -0,0 +1,33 @@
+@Table({
+  timestamps: false,
+  paranoid: true,
+  tableName: 'user_theme',
```
обычно названия таблиц пишут во мн. числе 'user_themes'

shchegol replied:
Да, бездумно скопировал с курса практикума

alexch365 commented on src/server/models/siteTheme.ts:
```diff
@@ -0,0 +1,32 @@
+@Table({
+  timestamps: false,
+  paranoid: true,
+  tableName: 'site_theme',
```
site_themes

shchegol committed:
Merge branch 'develop' of https://github.com/shchegol/reactive-battle into #105_theme-api

 Conflicts:
	src/server/routes/commentRouterFactory.ts
	src/server/routes/topicRouterFactory.ts
	src/server/server.ts

shchegol committed:
#105 theme api
+ добавлен auth на userTheme и themes routes
+ добавлен прямой путь к папке @server
* bugfix название баз в множ. числе
